web/src/lib/components/Navbar.svelte

```diff
@@ -71,24 +104,44 @@
               return;
           }
 
-          if (u.nym === null || u.nym === "") {
-              showModal(TwitterUsername, true,
-                  (saved) => {
-                      if (saved) {
-                          if ($user && !$user.twitterUsernameVerified) {
-                              showModal(TwitterVerification, true);
-                          }
-                      } else {
-                          // trying to hide the modal if you didn't set your Twitter username logges you out
-                          logout(false);
-                      }
-                  });
-          }
+          if (u.nostrPublicKey === null) {
+            if (u.nym === null || u.nym === "") {
+                showModal(TwitterUsername, true,
+                    (saved) => {
+                        if (saved) {
+                            if ($user && !$user.twitterUsernameVerified) {
+                                showModal(TwitterVerification, true);
+                            }
+                        } else {
+                            // trying to hide the modal if you didn't set your Twitter username logges you out
+                            logout();
+                        }
+                    });
+            }
+        } else {
+            if (u.nym === null || u.nym === "") {
+                let gotProfile = false;
+                pool.connectAndGetProfile(decodeNpub(u.nostrPublicKey),
+                    nostrProfile => {
+                        if (gotProfile) {
+                            return;
+                        }
+                        gotProfile = true;
+
+                        let name = <string>nostrProfile.name;
+                        name = name.replace(/[^a-zA-Z0-9]/g, "").toLowerCase();
+                        while (name.length < 3) {
+                            name += "0"; // just pas with zeroes - not ideal, but they can always change it later
+                        }
+
+                        saveProfile(name, nostrProfile.picture);
```

Comment:
I get that it's conditional, but connecting to a Nostr pool in Navbar sounds a little weird.

Besides, when is `u.nym` filled here so we stop connecting to Nostr?

Reply:
I agree it is weird. Will move to the server side, probably. Or somewhere else (main layout)? Can you think about some pros and cons of server-side versus client-side BTW?

About u.nym - saveProfile, on success, will set the global user store, which we subscribe to. Yeah, not the cleanest part of the front-end, I know, this part has been bothering me for quite a while, actually...

Reply:
It's not that I like it a lot more, but maybe inside `getProfile`, if you can call it asynchronously...

It's a little better place. The thing that is not improved is that it will be called a lot anyways, right? But maybe that's not a problem, since it's a client operation...

If you move it to the backend, you could try and get all the profiles at once each 24 hours like I do in `queryProfilesToNostrRelaysInBatches`... but that's a bigger change.